Riot/Modules/Room/Location/LocationMarkerView.swift

```diff
@@ -20,14 +20,23 @@ import Mapbox
 
 class LocationMarkerView: MGLAnnotationView, NibLoadable {
     
+    @IBOutlet private var markerBackground: UIImageView!
     @IBOutlet private var avatarView: UserAvatarView!
     
+    private static var usernameColorGenerator = UserNameColorGenerator()
+    private let theme: Theme = ThemeService.shared().theme
+    
     override func awakeFromNib() {
         super.awakeFromNib()
         translatesAutoresizingMaskIntoConstraints = false
     }
     
     func setAvatarData(_ avatarData: AvatarViewDataProtocol) {
+        Self.usernameColorGenerator.defaultColor = theme.colors.primaryContent
+        Self.usernameColorGenerator.userNameColors = theme.colors.namesAndAvatars
```

Comment:
As we are starting to use the `UserNameColorGenerator` in several places we can probably factorize the theming part like this in the `UserNameColorGenerator` class:
```
// MARK: - Themable
extension UserNameColorGenerator: Themable {
    
    func update(theme: Theme) {
        self.defaultColor = theme.colors.primaryContent
        self.userNameColors = theme.colors.namesAndAvatars
    }
}
```